packages/extension-dapp/README.md

```diff
@@ -35,4 +35,9 @@ api.setSigner(injector.signer);
 api.tx.balances
   .transfer('5C5555yEXUcmEJ5kkcCMvdZjUo7NGJiQJMS7vZXEeoMhj3VQ', 123456)
   .signAndSend('5DTestUPts3kjeXSTMyerHihn1uwMfLj8vU8sqF7qYrFabHE', (status) => { ... });
+
+// retrieve a RPC provider from a particular source
+// if the extension does not expose this provider, it will return an error.
+const provider = web3RpcProvider('polkadot-js', { type: 'WsProvider', payload: 'http://locahost:9944' });
```

Comment:
just a thought, but in the future maybe we should restrict the use of one `web3RpcProvider` (`setProvider`) per page, otherwise a page we allowed might DoS an endpoint if it keeps calling `setProvider` to the same endpoint (or variations thereof) to connect/disconnect to it repeatedly

Reply:
right, I went with the pre-configured providers way